### PR TITLE
Party menu app

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: src/party_context_menu.c [hunk 1]
```diff
@@ -0,0 +1,1290 @@
...
+            unk->active = FALSE;
+            return FALSE;
+        }
+        break;
+    }
+
+    return TRUE;
+}
+
+static void addFiveWindows(BgConfig *bgConfig, Window *window, const WindowTemplate *template) {
```
Comment (adrienntindall) [inline src/party_context_menu.c:380]: follow function naming conventions

File: src/party_context_menu.c [hunk 2]
```diff
@@ -0,0 +1,1290 @@
...
+    stats[0] = GetMonData(mon, MON_DATA_MAXHP, NULL);
+    stats[1] = GetMonData(mon, MON_DATA_ATK, NULL);
+    stats[2] = GetMonData(mon, MON_DATA_DEF, NULL);
+    stats[3] = GetMonData(mon, MON_DATA_SPATK, NULL);
+    stats[4] = GetMonData(mon, MON_DATA_SPDEF, NULL);
+    stats[5] = GetMonData(mon, MON_DATA_SPEED, NULL);
+    AddWindowParameterized(partyMenu->bgConfig, &partyMenu->levelUpStatsWindow[0], GF_BG_LYR_MAIN_0, 1, 1, 14, 12, 0, 0x260);
+    DrawFrameAndWindow1(&partyMenu->levelUpStatsWindow[0], TRUE, 1, 14);
+    FillWindowPixelBuffer(&partyMenu->levelUpStatsWindow[0], 15);
+    str_plusSign = NewString_ReadMsgData(partyMenu->msgData, msg_0300_00175);
```
Comment (adrienntindall) [inline src/party_context_menu.c:902]: dont use snake_case

File: src/party_context_menu.c [hunk 3]
```diff
@@ -0,0 +1,1290 @@
...
+            fillValue = 4;
+            color = MAKE_TEXT_COLOR(14, 15, 4);
+        } else {
+            fillValue = 11;
+            color = MAKE_TEXT_COLOR(14, 15, 11);
+        }
+        y = 4;
+        x = FontID_String_GetCenterAlignmentX(4, contextMenu->items[selection].text, 0, GetWindowWidth(&partyMenu->contextMenuButtonWindows[windowId]) * 8);
+    } else {
+        if (depressed == FALSE) {
```
Comment (adrienntindall) [inline src/party_context_menu.c:1020]: me when I take my meds

File: src/party_context_menu.c [hunk 4]
```diff
@@ -0,0 +1,1290 @@
...
+    } else {
+        Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_9], TRUE);
+    }
+    if (partyMenu->args->context == PARTY_MENU_CONTEXT_UNION_ROOM_BATTLE_SELECT || partyMenu->args->context == PARTY_MENU_CONTEXT_17 || partyMenu->args->context == PARTY_MENU_CONTEXT_BATTLE_HALL || partyMenu->args->context == PARTY_MENU_CONTEXT_23) {
+        Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_8], TRUE);
+    }
+}
+
+// NOTE: naming style reflective of that these do not operate on the central type `PartyMenuStruct *` of this TU
+static BOOL hitboxHasSelectionMapping(int state, int numItems, int selection) {
```
Comment (adrienntindall) [inline src/party_context_menu.c:1113]: still stay consistent with other function naming styles and keep the first letters capitalized

File: src/party_context_menu.c [hunk 5]
```diff
@@ -0,0 +1,1290 @@
...
+    for (int i = 0; i < 8; ++i) {
+        if (selection == sButtonWindowIDs[numItems - 1][state][i]) {
+            return TRUE;
+        }
+    }
+
+    return FALSE;
+}
+
+static int hitboxToSelection(int state, int numItems, int selection) {
```
Comment (adrienntindall) [inline src/party_context_menu.c:1123]: etc for remaining functions

File: src/party_menu.c [hunk 6]
```diff
@@ -0,0 +1,2723 @@
...
+}
+
+static u8 PartyMenu_IsMonDrawStateActive(PartyMenuStruct *partyMenu, u8 partySlot) {
+    return partyMenu->monsDrawState[partySlot].active;
+}
+
+BOOL sub_02079E38(PartyMenuStruct *partyMenu, u8 partySlot) {
+    Pokemon *mon;
+    u32 unused;
+    u16 species;  // sp4
```
Comment (adrienntindall) [inline src/party_menu.c:1018]: remove decomp comment

File: src/party_menu.c [hunk 7]
```diff
@@ -0,0 +1,2723 @@
...
+
+    (void)unused;
```
Comment (adrienntindall) [inline src/party_menu.c:1020]: can this not be removed?

File: src/party_menu.c [hunk 8]
```diff
@@ -0,0 +1,2723 @@
...
+        }
+    }
+
+    DrawFrameAndWindow2(&partyMenu->windows[PARTY_MENU_WINDOW_ID_34], TRUE, 0x02A, 15);
+    FillWindowPixelBuffer(&partyMenu->windows[PARTY_MENU_WINDOW_ID_34], 15);
+    PartyMenu_PrintBufferedMessageOnWindow34(partyMenu);
+    return result;
+}
+
+static int sub_0207C5D4(PartyMenuStruct *partyMenu, Pokemon *mon, s32 *transformResult) {
```
Comment (adrienntindall) [inline src/party_menu.c:2510]: SwapHeldItemGiratina?

File: src/party_menu.c [hunk 9]
```diff
@@ -0,0 +1,2723 @@
...
+        thunk_Sprite_SetPalIndex(partyMenu->sprites[PARTY_MENU_SPRITE_ID_CURSOR], 0);
+        partyMenu->args->context = PARTY_MENU_CONTEXT_0;
+        return PARTY_MENU_STATE_1;
+    } else {
+        partyMenu->args->selectedAction = PARTY_MENU_ACTION_RETURN_10;
+        return PARTY_MENU_STATE_BEGIN_EXIT;
+    }
+}
+
+static int sub_0207C908(PartyMenuStruct *partyMenu) {
```
Comment (adrienntindall) [inline src/party_menu.c:2627]: GiveHeldItem?

Comment (adrienntindall) [reply src/party_menu.c:2627]: Also PartyMenuStruct ->PartyMenu

File: src/party_menu.c [hunk 10]
```diff
@@ -0,0 +1,2723 @@
...
+        manipulateItemResult = PARTY_MENU_STATE_PRINT_GIVE_GRISEOUS_ORB_MESSAGE;
+    }
+    if (oldItem == ITEM_NONE) {
+        ReadMsgDataIntoString(partyMenu->msgData, msg_0300_00107, partyMenu->unformattedStrBuf);
+        BufferBoxMonNickname(partyMenu->msgFormat, 0, Mon_GetBoxMon(mon));
+        BufferItemName(partyMenu->msgFormat, 1, partyMenu->args->itemId);
+        StringExpandPlaceholders(partyMenu->msgFormat, partyMenu->formattedStrBuf, partyMenu->unformattedStrBuf);
+    } else {
+        Bag_AddItem(partyMenu->args->bag, oldItem, 1, HEAP_ID_PARTY_MENU);
+        ReadMsgDataIntoString(partyMenu->msgData, msg_0300_00085, partyMenu->unformattedStrBuf);
```
Comment (adrienntindall) [inline src/party_menu.c:2643]: Comment what messages read above pls (I prolly missed a ton this file so blanket comment for party_menu.c)

File: src/party_menu.c [hunk 11]
```diff
@@ -0,0 +1,2723 @@
...
+        PartyMenu_PrintMessageOnWindow34(partyMenu, msg_0300_00179, TRUE);
+        partyMenu->yesCallback = PartyMenu_ConfirmRemoveCapsuleYes;
+        partyMenu->noCallback = PartyMenu_ConfirmRemoveCapsuleNo;
+        partyMenu->afterTextPrinterState = PARTY_MENU_STATE_YES_NO_INIT;
+        return PARTY_MENU_STATE_WAIT_TEXT_PRINTER;
+    }
+}
+
+u32 sub_0207CA9C(void) {
+    return NARC_plist_gra_plist_gra_00000020_NCGR;
```
Comment (adrienntindall) [inline src/party_menu.c:2672]: what graphics are these? can prolly name the functions since you alr dumped them

File: src/party_menu_items.c [hunk 12]
```diff
@@ -0,0 +1,1015 @@
...
+    PartyMenu_DrawMonHpBarOnWindow(partyMenu, partyMenu->partyMonIndex);
+    if (hp == partyMenu->monsDrawState[partyMenu->partyMonIndex].hp) {
+        PartyMenu_PrintMessageOnWindow34(partyMenu, -1, TRUE);
+        partyMenu->itemUseCallback = PartyMenu_ItemUseFunc_WaitTextPrinterThenExit;
+    }
+    return PARTY_MENU_STATE_ITEM_USE_CB;
+}
+
+BOOL ItemId_IsReviveAll(u16 itemId) {
+    return !!GetItemAttr(itemId, ITEMATTR_REVIVE_ALL, HEAP_ID_PARTY_MENU);
```
Comment (adrienntindall) [inline src/party_menu_items.c:500]: !!
(just use a comparison lmao)

File: src/party_menu_items.c [hunk 13]
```diff
@@ -0,0 +1,1015 @@
...
+}
+
+static u8 PartyMenu_SacredAshGetNextMonId(PartyMenuStruct *partyMenu, u8 partySlot) {
+    if (partySlot >= PARTY_SIZE) {
+        return 0xFF;
```
Comment (adrienntindall) [inline src/party_menu_items.c:505]: PARTY_SLOT_NONE

File: src/party_menu_list_items.c [hunk 14]
```diff
@@ -0,0 +1,830 @@
...
+    }
+    PartyMenu_SetTopScreenSelectionPanelVisibility(partyMenu, FALSE);
+    sub_0207A7F4(partyMenu, partyMenu->softboiledDonorSlot);
+    PartyMenu_PrintMessageOnWindow32(partyMenu, msg_0300_00029, TRUE);
+}
+
+void sub_0207FC1C(PartyMenuStruct *partyMenu) {
+    u16 *bg2Tilemap;
+    u16 *bg1Tilemap;
+    s8 spC;
```
Comment (adrienntindall) [inline src/party_menu_list_items.c:356]: try to not use sp terminology once the func is decompiled

File: src/party_menu_list_items.c [hunk 15]
```diff
@@ -0,0 +1,830 @@
...
+        partyMenu->levelUpStatsTmp[3] = PARTY_MENU_ACTION_RETURN_SOFTBOILED - PARTY_MENU_ACTION_RETURN_FIELD_MOVE_BEGIN;
+    }
+}
+
+static void PartyMonContextMenuAction_Headbutt(PartyMenuStruct *partyMenu, int *pState) {
+    partyMenu->args->selectedAction = PARTY_MENU_ACTION_RETURN_HEADBUTT;
+    PartyMenuContextMenuAction_FieldMoveCommon(partyMenu, pState);
+}
+
+static int sub_02080A58(PartyMenuStruct *partyMenu) {
```
Comment (adrienntindall) [inline src/party_menu_list_items.c:800]: Menu action for softboiled?

File: src/party_menu_list_items.c [hunk 16]
```diff
@@ -0,0 +1,830 @@
...
+    Set2dSpriteVisibleFlag(partyMenu->sprites[PARTY_MENU_SPRITE_ID_SWITCH_MON_CURSOR], TRUE);
+    sub_0207A7F4(partyMenu, partyMenu->softboiledDonorSlot);
+    PartyMenu_PrintMessageOnWindow32(partyMenu, msg_0300_00037, TRUE);
+    partyMenu->levelUpStatsTmp[1] = 0;
+    return PARTY_MENU_STATE_SOFTBOILED;
+}
+
+int sub_02080B74(PartyMenuStruct *partyMenu) {
+    ClearFrameAndWindow2(&partyMenu->windows[PARTY_MENU_WINDOW_ID_34], TRUE);
+    PartyMenu_PrintMessageOnWindow32(partyMenu, msg_0300_00029, TRUE);
```
Comment (adrienntindall) [inline src/party_menu_list_items.c:825]: messages for this file too, just leave them as a comment above

File: src/start_menu.c [hunk 17]
```diff
@@ -886,7 +886,7 @@ BOOL Task_StartMenu_HandleReturn_Pokemon(TaskManager *taskManager) {
         pokemonSummaryArgs->natDexEnabled = SaveArray_IsNatDexEnabled(fieldSystem->saveData);
         pokemonSummaryArgs->unk2C = sub_02088288(fieldSystem->saveData);
         pokemonSummaryArgs->unk28 = 0;
-        pokemonSummaryArgs->unk30 = &fieldSystem->unk_10C;
+        pokemonSummaryArgs->unk30 = &fieldSystem->menuInputState;
```
Comment (adrienntindall) [inline src/start_menu.c:889]: unk30

File: src/start_menu.c [hunk 18]
```diff
@@ -967,69 +967,69 @@ BOOL Task_StartMenu_HandleReturn_Pokemon(TaskManager *taskManager) {
...
-    case 8: {
+    case PARTY_MENU_ACTION_RETURN_EVO_ITEM_USE: {
         UnkStruct_0203CA9C_Case8 *unk = AllocFromHeap(HEAP_ID_FIELD, sizeof(UnkStruct_0203CA9C_Case8));
         unk->evoParam = partyMenuArgs->itemId;
-        unk->evoType = 3;
+        unk->evoType = EVOCTX_ITEM_USE;
         unk->partySlot = partyMenuArgs->partySlot;
         unk->species = partyMenuArgs->species;
-        unk->unk_8 = partyMenuArgs->unk_40;
+        unk->unk_8 = partyMenuArgs->evoMethod;
```
Comment (adrienntindall) [inline src/start_menu.c:981]: unk_8

File: src/unk_0206B910.c [hunk 19]
```diff
@@ -100,7 +100,7 @@ static BOOL sub_0206B984(TaskManager *taskManager) {
         SealCase *sealCase = Save_SealCase_Get(r7->saveData);
         r6->sealCase = sealCase;
         r6->unk31 = 0;
-        r6->unk2c = &fieldSystem->unk_10C;
+        r6->unk2c = &fieldSystem->menuInputState;
```
Comment (adrienntindall) [inline src/unk_0206B910.c:103]: unk2C

File: src/unk_02092BE8.c [hunk 20]
```diff
@@ -54,7 +54,7 @@ static void sub_02092BE8(FieldSystem* sys, PokegearArgs* ptr, BOOL a2) {
     }
 
     ptr->playerGender = PlayerProfile_GetTrainerGender(Save_PlayerData_GetProfileAddr(FieldSystem_GetSaveData(sys)));
-    ptr->unk28 = &sys->unk_10C;
+    ptr->unk28 = &sys->menuInputState;
```
Comment (adrienntindall) [inline src/unk_02092BE8.c:57]: unk28

**Review by adrienntindall — CHANGES_REQUESTED** [review]

File: src/party_context_menu.c (hunk 1 above)
Comment (PikalaxALT) [reply src/party_context_menu.c:380]: exception to the convention is local functions that don't operate on the file's primary object

File: src/party_context_menu.c (hunk 2 above)
Comment (PikalaxALT) [reply src/party_context_menu.c:902]: this is a valid use case

File: src/party_context_menu.c (hunk 4 above)
Comment (PikalaxALT) [reply src/party_context_menu.c:1113]: i'd rather have this visual distinction for these kinds of helpers

File: src/party_menu.c (hunk 7 above)
Comment (PikalaxALT) [reply src/party_menu.c:1020]: necessary to match. don't ask why.

File: src/party_menu.c (hunk 10 above)
Comment (PikalaxALT) [reply src/party_menu.c:2643]: Most of these are just {PLACEHOLDER} so nothing really interesting here. If people want to look them up, they can cross-reference the GMM. Commenting what the messages say inside the C code creates an opportunity for desync.

File: src/party_menu.c (hunk 11 above)
Comment (PikalaxALT) [reply src/party_menu.c:2672]: no fucking clue lol

File: src/party_menu_items.c (hunk 13 above)
Comment (PikalaxALT) [reply src/party_menu_items.c:505]: not creating a whole constant just for this case

File: src/party_menu_list_items.c (hunk 14 above)
Comment (PikalaxALT) [reply src/party_menu_list_items.c:356]: not a big enough deal to fix, as they'll be renamed once the function is better understood

File: include/overlay_94.h [hunk 21]
```diff
@@ -13,8 +13,8 @@ typedef struct UnkStruct_Overlay_94_B {
     u8 unk27;
 } UnkStruct_Overlay_94_B;
 
-void PartyMenu_InitIconFormChangeData(PartyMenuStruct* unkPtr);
-BOOL PartyMenu_AnimateIconFormChange(PartyMenuStruct* unkPtr);
+void PartyMenu_InitIconFormChangeData(PartyMenu* unkPtr);
```
Comment (adrienntindall) [inline include/overlay_94.h:16]: unkPtr -> partyMenu

File: include/unk_02092BE8.h [hunk 22]
```diff
@@ -29,7 +29,7 @@ typedef struct PokegearArgs{
     u8 matrixXCoord; //0x22
     u8 matrixYCoord; //0x23
     SaveData* saveData; //0x24
-    void *unk28; //Pointer to a Field System variable at offset 0x10C (currently unk_10C[0];)
+    void *menuInputStatePtr; //Pointer to a Field System variable at offset 0x10C (currently unk_10C[0];)
```
Comment (adrienntindall) [inline include/unk_02092BE8.h:32]: can remove comment now

File: src/overlay_94.c [hunk 23]
```diff
@@ -20,18 +20,18 @@
 #include "msgdata/msg/msg_0300.h"
 #include "overlay_94.h"
 
-static void _DestroyLocalWork(struct PartyMenuStruct *unkPtr);
-static void _InitEffects(struct PartyMenuStruct *unkPtr);
-static void _CleanupEffects(struct PartyMenuStruct *unkPtr);
+static void _DestroyLocalWork(PartyMenu *unkPtr);
```
Comment (adrienntindall) [inline src/overlay_94.c:23]: unkPtr -> partyMenu

File: src/party_context_menu.c [hunk 24]
```diff
@@ -431,7 +431,7 @@ void PartyMenu_RemoveAllWindows(PartyMenuStruct *partyMenu) {
     }
 }
 
-void PartyMenu_SetContextMenuStaticStrings(PartyMenuStruct *partyMenu) {
+void PartyMenu_SetContextMenuStaticStrings(PartyMenu *partyMenu) {
     ReadMsgDataIntoString(partyMenu->msgData, msg_0300_00128, partyMenu->contextMenuStrings[PARTY_MON_CONTEXT_MENU_SWITCH]);
```
Comment (adrienntindall) [inline src/party_context_menu.c:435]: I still think it's a good idea to comment what the messages are but I can do that in a separate PR later if you don't wanna do it rn

File: src/voltorb_flip/voltorb_flip.c [hunk 25]
```diff
@@ -1970,7 +1970,7 @@ static void ov122_021E8094(OVY_MANAGER *man) {
     work->heapId = HEAP_ID_VOLTORB_FLIP;
     work->options = args->options;
     work->coins = args->coins;
-    work->unkC = args->unk8;
+    work->unkC = args->menuInputStatePtr;
```
Comment (adrienntindall) [inline src/voltorb_flip/voltorb_flip.c:1973]: unkC -> menuInputStatePtr

**Review by adrienntindall — DISMISSED** [review]
Few very minor things, just go ahead and address em and feel free to merge right after

**Comment by PikalaxALT** [comment]
> Few very minor things, just go ahead and address em and feel free to merge right after

ok but you know that as soon as i push these changes it'll dismiss your review

**Review by adrienntindall — APPROVED** [review]

**Pull request merged**